### ISSUE TITLE
Files being changed even when all rules are met and no changes are made

### ISSUE THREAD

**Issue by kowsheek** [editorconfig/editorconfig-vscode#94]
### Steps to reproduce

Editor config:

```ini
root = true
[*]
indent_style = space
indent_size = 4
end_of_line = lf
charset = utf-8
trim_trailing_whitespace = true
insert_final_newline = true
[*.md]
trim_trailing_whitespace = false
```

Versions:
EditorConfig extension: `0.3.4`
VSCode: `1.8.1`

Steps:
* Open a file
* Being asked to save

### Expected Behavior

Should not be asked to save.

### Actual Behavior

The file is being asked to save.

**Comment by jedmao** [comment]
Is it possible you have mixed lined endings? vscode doesn't let you have mixed line endings so it might appear as though all is well, but in fact it's not.

Also, did you try disabling any/all plugins and see if you have the same experience? If not, try enabling just the vscode extension and see if it presents itself, then report back.

Possibly related: https://github.com/editorconfig/editorconfig-vscode/issues/57

**Comment by kowsheek** [comment]
I'm not sure about the mixed line endings. I see `LF` on the files before and after opening it.

But, once I disable the plugin, this issue doesn't occur.

**Comment by jedmao** [comment]
When you say you're being asked to save, you mean the file is marked dirty, right?

Well, the file won't be marked dirty for no reason, so you'll have to identify what's changing; thus, why the file is dirty. I think we can safely assume the file contents are not changing. We can also assume `insert_final_newline` is not being applied on file open, because that's a save operation. The only thing left is line endings and charset, but the extension doesn't yet support charset, so it has to be line endings, right?

I'm afraid you'll need to do some more testing on your end to identify the exact thing that's making the file dirty.

Also, on which operating system are you?

You could also just save the file on file open and see what git says changed in a diff. Just throwing out ideas. Maybe save a backup of the file before you save it though, just to make sure you know what's going on.

**Comment by sryze** [comment]
I'm having this problem too. When I open any file and try to close it without making any changes, I get asked whether I want to save changes as if the file was modified. 

If I disable EditorConfig this doesn't happen.

**Comment by SamVerschueren** [comment]
This is almost 100% certain due to the fact you are changing code that was committed on a Unix system, on a Windows system where git changes the line endings for you on checkout to LFCR and EditorConfig changes it back to LF

**Comment by sryze** [comment]
I was able to solve my particular problem by cloning the repo again. I think it happened because I added .gitattributes file at some point but files still were checked out with CRLF line endings (don't know why, I did do `git reset --hard`).

**Comment by jedmao** [comment]
@kowsheek how about you? Did you resolve your problem yet?

**Comment by kowsheek** [comment]
Hey @SamVerschueren & @jedmao! You were spot on, it was actually the line endings. I am on Windows and some of my other teammates are on MacOS so we were going with LF and was working fine but for some reason recently this has been acting up.

What do you recommend for cross-OS

**Comment by jedmao** [comment]
@kowsheek some Windows Git distributions (e.g., [GitHub for Windows](https://windows.github.com/)) configure your system to automatically convert line endings from `LF` to `CRLF` when files are checked out. This line-ending conversion is controlled by a Git configuration flag: `autocrlf`.

There is some controversy regarding the best value for `autocrlf`; however, having `CRLF` line endings will cause issues like the one you've just experienced. As such, I recommend configuring your repositories such that they never convert line endings.

To disable automatic line-ending conversion, you should visit _each_ of your repositories and execute the following commands:

```
git config core.autocrlf false
git rm --cached -r .
git reset --hard
```

The first command tells Git to never change line endings (in the future). Next, we refresh each repository by removing every file from Git's index and, finally, rewriting the Git index to pick up all the new line endings. This fixes the CRLFs that were introduced to your local file system when you cloned each repository.

Now, you can work in LF mode via EditorConfig and never introduce rogue CRLFs into your files.

You can also run `git config core.autocrlf false --system` to change the system-wide setting, which is actually what I do on Windows so I don't have to deal with this on a regular basis.

**Comment by kowsheek** [comment]
Amazing, thanks a lot @jedmao! I used your instructions and the ones here `http://stackoverflow.com/questions/2517190/how-do-i-force-git-to-use-lf-instead-of-crlf-under-windows` to get the repo to consistently use LF.

Thanks everyone for all the effort into this plugin as well, it's making my team's life WAY easier 🥇

**Comment by jedmao** [comment]
Thanks @kowsheek. I'm glad that worked out for you! In my experience, removing the `.gitattributes` file entirely and relying on the system setting is what I needed.

**_Note: all my team is on Mac except for me._**

**Comment by kowsheek** [comment]
I didn't need to remove `.gitattributes` which I use in my repo to specify binaries such as favicons. I did the global and local setting changes to get it rolling.

**Comment by jedmao** [comment]
@kowsheek fixed it properly in #116, published as v0.6.1.
